dockers/docker-fpm-frr/zsocket.sh

```diff
@@ -23,11 +23,10 @@ while getopts ":a:o:h" opt; do
             ;;
     esac
 done
-shift $((OPTIND-1))
 
 timeout 5s bash -c -- "until </dev/tcp/${addr}/${port}; do sleep 0.1;done"
```

Comment:
Can you please explain me what this does ? Does it wait only for 5seconds. Do you think it will be sufficient. In my case, I was waiting more because there may be a lot of configurations. What is port 2601. My command was
zebra_ready=$(netstat -tulpn | grep LISTEN | grep zebra)

Reply:
There is a script src/sonic-bgpcfgd/bgpcfgd/main.py where we wait for 20 seconds for all daemons to come up. We are creating a dependency here which will increase the time for daemons to come up. Hence, we should increase the time for all daemons to come up there in frr.wait_for_daemons().

Reply:
what is the meaning of line shift $((OPTIND-1))

Reply:
This line checks if the port is opened. Port 2601 is the default port for zebra connections. And it waits for at most 5 seconds for the port to be open. From my observation, the wait time for the port is about 0.2 seconds. So I think the 5 seconds limit should be enough. As this dependency only takes a limited amount of time, I do not think it is very necessary to extend the 20-second wait time for bgpcfgd (my observation of the wait time is about 2 seconds).

Reply:
For the line `shift $((OPTIND-1))`, the script was originally written for bgpd only. The options to this script were passed in along with the ones for bgpd. Therefore, we needed to shift off the processed options and only pass the ones for bgpd to bgpd. Since the script is now pulled out as an independent one and it is no longer necessary to pass the options, I removed the line.